backbone/src/main/java/org/researchstack/backbone/ui/step/layout/ActiveStepLayout.java

```diff
@@ -189,22 +193,44 @@ protected void start() {
         }
 
         recorderList = new ArrayList<>();
+        File outputDir = getOutputDirectory();
+
+        if (activeStep.getRecorderConfigurationList() != null) {
+            for (RecorderConfig config : getRecorderConfigurationList()) {
+                Recorder recorder = config.recorderForStep(activeStep, outputDir);
+                // recorder can be null if it requires custom setup
+                if (recorder == null) {
+                    recorder = createCustomRecorder(config);
```

Comment:
I added custom recorder setup functionality using an overridable class.  This is needed for the heart rate camera recorder, since it passes in a SurfaceView for the camera to attach to that is required for the recorder.